### PR TITLE
[fix](meta) add sync new image timeout

### LINKED ISSUE #25764: [Bug] sync image file timeout when adding new follower node
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0.2

### What's Wrong?

When adding new follower node to cluster, the follower sync image from FE master timeout.

The cluster meta image is about 1G. By testing, it takes about 20s to synchronizing image files

The error log:
`2023-10-23 09:35:57,278 WARN (main|1) [Env.getClusterIdAndRole():1041] current node is not added to the group. please add it first. sleep 5 seconds and retry, current helper nodes: [HostInfo{host='10.153.123.74', port=9010}]
2023-10-23 09:36:02,283 INFO (main|1) [Env.getFeNodeTypeAndNameFromHelpers():1198] get fe node type FOLLOWER, name fe_6aa20a8a_8380_4a27_9c65_f7f343cb8968 from 10.153.123.74:10.153.123.74:8030
2023-10-23 09:36:07,787 WARN (main|1) [DorisFE.start():209]
java.io.IOException: java.net.SocketTimeoutException: Read timed out
        at org.apache.doris.catalog.Env.getNewImage(Env.java:1664) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.getClusterIdAndRole(Env.java:1126) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.initialize(Env.java:884) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.DorisFE.start(DorisFE.java:163) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.DorisFE.main(DorisFE.java:83) ~[doris-fe.jar:1.2-SNAPSHOT]
Caused by: java.net.SocketTimeoutException: Read timed out
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_121]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1926) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1921) ~[?:1.8.0_121]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1920) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1490) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474) ~[?:1.8.0_121]
        at org.apache.doris.master.MetaHelper.getResponse(MetaHelper.java:130) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.master.MetaHelper.getRemoteFile(MetaHelper.java:98) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.getNewImage(Env.java:1657) ~[doris-fe.jar:1.2-SNAPSHOT]
        ... 4 more
Caused by: java.net.SocketTimeoutException: Read timed out
        at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_121]
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_121]
        at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_121]
        at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_121]
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246) ~[?:1.8.0_121]
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286) ~[?:1.8.0_121]
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345) ~[?:1.8.0_121]
        at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:704) ~[?:1.8.0_121]
        at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:647) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1569) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474) ~[?:1.8.0_121]
        at sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:2982) ~[?:1.8.0_121]
        at org.apache.doris.master.MetaHelper.getRemoteFile(MetaHelper.java:93) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.catalog.Env.getNewImage(Env.java:1657) ~[doris-fe.jar:1.2-SNAPSHOT]
        ... 4 more`

### What You Expected?

The new follower can synchronize image file from master

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CanGuan**
Add timeout config for sync

## Proposed changes

Issue Number: close #25764

Add sync_new_image_timeout config, for adjusting synchronize image timeout from master 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Have you try it? can it really solve the timeout issue?

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2259,4 +2259,13 @@ public class Config extends ConfigBase {
...
     public static boolean ignore_unknown_metadata_module = false;
 
+    @ConfField(mutable = true, masterOnly = true, description = {
+        "FE扩容时，从主节点同步元数据的timeout时间，根据${meta_dir}/image文件夹下面的image文件大小和节点间的网络环境调整，" +
+            "单位为毫秒，默认值30000",
+        "The timeout for new FE Follower/Observer synchronizing metadata from the FE Master, " +
+            "adjust by the size of image file in the ${meta_dir}/image and the network environment between nodes. " +
+            "The default values is 3000 ms."
+    })
+    public static int sync_new_image_timeout = 30000;
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2269]:
```suggestion
    public static int sync_new_image_timeout_s = 300;
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -1726,7 +1726,7 @@ private void getNewImage(HostInfo helperNode) throws IOException {
                 String url = "http://" + hostPort + "/image?version=" + version;
                 String filename = Storage.IMAGE + "." + version;
                 File dir = new File(this.imageDir);
-                MetaHelper.getRemoteFile(url, HTTP_TIMEOUT_SECOND * 1000, MetaHelper.getFile(filename, dir));
+                MetaHelper.getRemoteFile(url, Config.sync_new_image_timeout, MetaHelper.getFile(filename, dir));
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:1729]:
```suggestion
                MetaHelper.getRemoteFile(url, Config.sync_new_image_timeout_s * 1000, MetaHelper.getFile(filename, dir));
```

Comment (CanGuan) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java]: done

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java (hunk 1 above)
Comment (CanGuan) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java]: done

**Comment by CanGuan** [comment]
> Have you try it? can it really solve the timeout issue?

We found this bug by adding new follower to our doris cluster. By tracing this bug, we found the sync image timeout is only 5s and no way to modify it. After fixing and deploying to our cluster, it fix the problem. Here are the success log

image
In our case (image file is 1G),  Follower takes 22s to synchronize the file from FE master.

refer to issue https://github.com/apache/doris/issues/25764

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 3]
```diff
@@ -2259,4 +2259,13 @@ public class Config extends ConfigBase {
...
     public static boolean ignore_unknown_metadata_module = false;
 
+    @ConfField(mutable = true, masterOnly = true, description = {
+        "FE扩容时，从主节点同步元数据的timeout时间，根据${meta_dir}/image文件夹下面的image文件大小和节点间的网络环境调整，"
+            + "单位为秒，默认值300",
+        "The timeout for new FE Follower/Observer synchronizing metadata from the FE Master, "
+            + "adjust by the size of image file in the ${meta_dir}/image and the network environment between nodes. "
+            + "The default values is 300s."
+    })
+    public static int sync_new_image_timeout_s = 300;
```
Comment (dataroaring) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2269]: sync_image_timeout_second is a better name.

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java (hunk 3 above)
Comment (CanGuan) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java]: done

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by CanGuan** [comment]
run buildall

**Comment by CanGuan** [comment]
> LGTM

Please review again, chore code for checkstyle

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**